### ISSUE TITLE
Long virtual text from texlab prevents 'k' from navigating a line up

### ISSUE THREAD

**Issue by hnorkowski** [helix-editor/helix#9287]
### Summary

When referencing a figure in latex the caption is printed as virtual text by texlab. If this caption is long it prevents the navigation k to work. the cursor gets suck on the line with virtual text.

### Reproduction Steps

 
https://asciinema.org/a/wrt3gD96COuSDZLE0mfgGiPwj

I tried this:

`test.tex`:
```tex
\documentclass[11pt]{article}

\begin{document}
	Sample \ref{fig:lorem} Figure
	\begin{figure}
		\caption{Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.}\label{fig:lorem}
	\end{figure}
\end{document}
```

1. `hx test.tex`
2. navigating downwards using 'j'
3. navigating upwards using 'k'

I expected this to happen: Get all the way up to the top of the document

Instead, this happened: The cursor gets stuck at the end of the virtual text. Pressing 'h' moves the cursor before the virtual text.

LSP Version: texlab 5.12.1

### Helix log

<details><summary>~/.cache/helix/helix.log</summary>

```
2024-01-09T11:45:34.752 helix_view::clipboard [DEBUG] Using xclip to interact with the system and selection (primary) clipboard
2024-01-09T11:45:34.883 helix_vcs [INFO] Error {
    context: "failed to open git repo",
    source: Discover(
        NoGitRepository {
            path: "/home/nohandrick/tmp",
        },
    ),
}
2024-01-09T11:45:34.883 helix_vcs [INFO] failed to open diff base for /home/nohandrick/tmp/test.tex
2024-01-09T11:45:34.883 helix_vcs [INFO] Error {
    context: "failed to open git repo",
    source: Discover(
        NoGitRepository {
            path: "/home/nohandrick/tmp",
        },
    ),
}
2024-01-09T11:45:34.883 helix_vcs [INFO] failed to obtain current head name for /home/nohandrick/tmp/test.tex
2024-01-09T11:45:34.883 helix_lsp::client [INFO] Using custom LSP config: {"texlab":{"build":{"args":["-pdf","-interaction=nonstopmode","-synctex=1","%f"],"executable":"latexmk","forwardSearchAfter":true,"onSave":true},"chktex":{"onEdit":true,"onOpenAndSave":true},"forwardSearch":{"args":["--synctex-forward","%l:1:%f","%p"],"executable":"zathura"}}}
2024-01-09T11:45:34.883 helix_lsp::transport [INFO] texlab -> {"jsonrpc":"2.0","method":"initialize","params":{"capabilities":{"general":{"positionEncodings":["utf-8","utf-32","utf-16"]},"textDocument":{"codeAction":{"codeActionLiteralSupport":{"codeActionKind":{"valueSet":["","quickfix","refactor","refactor.extract","refactor.inline","refactor.rewrite","source","source.organizeImports"]}},"dataSupport":true,"disabledSupport":true,"isPreferredSupport":true,"resolveSupport":{"properties":["edit","command"]}},"completion":{"completionItem":{"deprecatedSupport":true,"insertReplaceSupport":true,"resolveSupport":{"properties":["documentation","detail","additionalTextEdits"]},"snippetSupport":true,"tagSupport":{"valueSet":[1]}},"completionItemKind":{}},"hover":{"contentFormat":["markdown"]},"inlayHint":{"dynamicRegistration":false},"publishDiagnostics":{"versionSupport":true},"rename":{"dynamicRegistration":false,"honorsChangeAnnotations":false,"prepareSupport":true},"signatureHelp":{"signatureInformation":{"activeParameterSupport":true,"documentationFormat":["markdown"],"parameterInformation":{"labelOffsetSupport":true}}}},"window":{"workDoneProgress":true},"workspace":{"applyEdit":true,"configuration":true,"didChangeConfiguration":{"dynamicRegistration":false},"didChangeWatchedFiles":{"dynamicRegistration":true,"relativePatternSupport":false},"executeCommand":{"dynamicRegistration":false},"fileOperations":{"didRename":true,"willRename":true},"inlayHint":{"refreshSupport":false},"symbol":{"dynamicRegistration":false},"workspaceEdit":{"documentChanges":true,"failureHandling":"abort","normalizesLineEndings":false,"resourceOperations":["create","rename","delete"]},"workspaceFolders":true}},"clientInfo":{"name":"helix","version":"23.10 (783ff27b)"},"initializationOptions":{"texlab":{"build":{"args":["-pdf","-interaction=nonstopmode","-synctex=1","%f"],"executable":"latexmk","forwardSearchAfter":true,"onSave":true},"chktex":{"onEdit":true,"onOpenAndSave":true},"forwardSearch":{"args":["--synctex-forward","%l:1:%f","%p"],"executable":"zathura"}}},"processId":1433864,"rootPath":"/home/nohandrick/tmp","rootUri":null,"workspaceFolders":[]},"id":0}
2024-01-09T11:45:34.883 helix_lsp::client [INFO] Using custom LSP config: {"ltex":{"dictionary":{<censored>}}}
2024-01-09T11:45:34.883 helix_lsp::transport [INFO] ltex-ls -> {"jsonrpc":"2.0","method":"initialize","params":{"capabilities":{"general":{"positionEncodings":["utf-8","utf-32","utf-16"]},"textDocument":{"codeAction":{"codeActionLiteralSupport":{"codeActionKind":{"valueSet":["","quickfix","refactor","refactor.extract","refactor.inline","refactor.rewrite","source","source.organizeImports"]}},"dataSupport":true,"disabledSupport":true,"isPreferredSupport":true,"resolveSupport":{"properties":["edit","command"]}},"completion":{"completionItem":{"deprecatedSupport":true,"insertReplaceSupport":true,"resolveSupport":{"properties":["documentation","detail","additionalTextEdits"]},"snippetSupport":true,"tagSupport":{"valueSet":[1]}},"completionItemKind":{}},"hover":{"contentFormat":["markdown"]},"inlayHint":{"dynamicRegistration":false},"publishDiagnostics":{"versionSupport":true},"rename":{"dynamicRegistration":false,"honorsChangeAnnotations":false,"prepareSupport":true},"signatureHelp":{"signatureInformation":{"activeParameterSupport":true,"documentationFormat":["markdown"],"parameterInformation":{"labelOffsetSupport":true}}}},"window":{"workDoneProgress":true},"workspace":{"applyEdit":true,"configuration":true,"didChangeConfiguration":{"dynamicRegistration":false},"didChangeWatchedFiles":{"dynamicRegistration":true,"relativePatternSupport":false},"executeCommand":{"dynamicRegistration":false},"fileOperations":{"didRename":true,"willRename":true},"inlayHint":{"refreshSupport":false},"symbol":{"dynamicRegistration":false},"workspaceEdit":{"documentChanges":true,"failureHandling":"abort","normalizesLineEndings":false,"resourceOperations":["create","rename","delete"]},"workspaceFolders":true}},"clientInfo":{"name":"helix","version":"23.10 (783ff27b)"},"initializationOptions":{"ltex":{"dictionary":{censored}}},"processId":1433864,"rootPath":"/home/nohandrick/tmp","rootUri":null,"workspaceFolders":[]},"id":0}
2024-01-09T11:45:34.883 helix_view::editor [DEBUG] editor status: Loaded 1 file.
2024-01-09T11:45:34.883 helix_lsp::client [INFO] Using custom LSP config: {"max-completion_items":20,"snippets_first":true}
2024-01-09T11:45:34.883 helix_lsp::transport [INFO] scls -> {"jsonrpc":"2.0","method":"initialize","params":{"capabilities":{"general":{"positionEncodings":["utf-8","utf-32","utf-16"]},"textDocument":{"codeAction":{"codeActionLiteralSupport":{"codeActionKind":{"valueSet":["","quickfix","refactor","refactor.extract","refactor.inline","refactor.rewrite","source","source.organizeImports"]}},"dataSupport":true,"disabledSupport":true,"isPreferredSupport":true,"resolveSupport":{"properties":["edit","command"]}},"completion":{"completionItem":{"deprecatedSupport":true,"insertReplaceSupport":true,"resolveSupport":{"properties":["documentation","detail","additionalTextEdits"]},"snippetSupport":true,"tagSupport":{"valueSet":[1]}},"completionItemKind":{}},"hover":{"contentFormat":["markdown"]},"inlayHint":{"dynamicRegistration":false},"publishDiagnostics":{"versionSupport":true},"rename":{"dynamicRegistration":false,"honorsChangeAnnotations":false,"prepareSupport":true},"signatureHelp":{"signatureInformation":{"activeParameterSupport":true,"documentationFormat":["markdown"],"parameterInformation":{"labelOffsetSupport":true}}}},"window":{"workDoneProgress":true},"workspace":{"applyEdit":true,"configuration":true,"didChangeConfiguration":{"dynamicRegistration":false},"didChangeWatchedFiles":{"dynamicRegistration":true,"relativePatternSupport":false},"executeCommand":{"dynamicRegistration":false},"fileOperations":{"didRename":true,"willRename":true},"inlayHint":{"refreshSupport":false},"symbol":{"dynamicRegistration":false},"workspaceEdit":{"documentChanges":true,"failureHandling":"abort","normalizesLineEndings":false,"resourceOperations":["create","rename","delete"]},"workspaceFolders":true}},"clientInfo":{"name":"helix","version":"23.10 (783ff27b)"},"initializationOptions":{"max-completion_items":20,"snippets_first":true},"processId":1433864,"rootPath":"/home/nohandrick/tmp","rootUri":null,"workspaceFolders":[]},"id":0}
2024-01-09T11:45:34.884 helix_tui::backend::crossterm [DEBUG] The keyboard enhancement protocol is not supported in this terminal (checked in 464.821µs)
2024-01-09T11:45:34.885 helix_term::application [DEBUG] received editor event: IdleTimer
2024-01-09T11:45:34.885 helix_lsp::transport [INFO] texlab <- {"jsonrpc":"2.0","id":0,"result":{"capabilities":{"completionProvider":{"resolveProvider":true,"triggerCharacters":["\\","{","}","@","/"," "]},"definitionProvider":true,"documentFormattingProvider":true,"documentHighlightProvider":true,"documentLinkProvider":{"resolveProvider":false},"documentSymbolProvider":true,"executeCommandProvider":{"commands":["texlab.cleanAuxiliary","texlab.cleanArtifacts","texlab.changeEnvironment","texlab.findEnvironments","texlab.showDependencyGraph","texlab.cancelBuild"]},"foldingRangeProvider":true,"hoverProvider":true,"inlayHintProvider":true,"referencesProvider":true,"renameProvider":{"prepareProvider":true},"textDocumentSync":{"change":2,"openClose":true,"save":{"includeText":false}},"workspaceSymbolProvider":true},"serverInfo":{"name":"TexLab","version":"5.12.1"}}}
2024-01-09T11:45:34.885 helix_lsp::transport [INFO] texlab <- {"capabilities":{"completionProvider":{"resolveProvider":true,"triggerCharacters":["\\","{","}","@","/"," "]},"definitionProvider":true,"documentFormattingProvider":true,"documentHighlightProvider":true,"documentLinkProvider":{"resolveProvider":false},"documentSymbolProvider":true,"executeCommandProvider":{"commands":["texlab.cleanAuxiliary","texlab.cleanArtifacts","texlab.changeEnvironment","texlab.findEnvironments","texlab.showDependencyGraph","texlab.cancelBuild"]},"foldingRangeProvider":true,"hoverProvider":true,"inlayHintProvider":true,"referencesProvider":true,"renameProvider":{"prepareProvider":true},"textDocumentSync":{"change":2,"openClose":true,"save":{"includeText":false}},"workspaceSymbolProvider":true},"serverInfo":{"name":"TexLab","version":"5.12.1"}}
2024-01-09T11:45:34.885 helix_lsp::transport [INFO] texlab -> {"jsonrpc":"2.0","method":"initialized","params":{}}
2024-01-09T11:45:34.885 helix_term::application [DEBUG] received editor event: LanguageServerMessage((0, Notification(Notification { jsonrpc: None, method: "initialized", params: None })))
2024-01-09T11:45:34.885 helix_lsp::transport [INFO] texlab -> {"jsonrpc":"2.0","method":"workspace/didChangeConfiguration","params":{"settings":{"texlab":{"build":{"args":["-pdf","-interaction=nonstopmode","-synctex=1","%f"],"executable":"latexmk","forwardSearchAfter":true,"onSave":true},"chktex":{"onEdit":true,"onOpenAndSave":true},"forwardSearch":{"args":["--synctex-forward","%l:1:%f","%p"],"executable":"zathura"}}}}}
2024-01-09T11:45:34.885 helix_lsp::transport [INFO] texlab -> {"jsonrpc":"2.0","method":"textDocument/didOpen","params":{"textDocument":{"languageId":"latex","text":"\\documentclass[11pt]{article}\n\n\\begin{document}\n\tSample \\ref{fig:lorem} Figure\n\t\\begin{figure}\n\t\t\\caption{Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.}\\label{fig:lorem}\n\t\\end{figure}\n\\end{document}\n","uri":"file:///home/nohandrick/tmp/test.tex","version":0}}}
2024-01-09T11:45:34.885 helix_lsp::transport [INFO] texlab <- {"jsonrpc":"2.0","id":1,"method":"workspace/configuration","params":{"items":[{"section":"texlab"}]}}
2024-01-09T11:45:34.885 helix_term::application [DEBUG] received editor event: LanguageServerMessage((0, MethodCall(MethodCall { jsonrpc: Some(V2), method: "workspace/configuration", params: Map({"items": Array [Object {"section": String("texlab")}]}), id: Num(1) })))
2024-01-09T11:45:34.885 helix_lsp::transport [INFO] texlab <- {"jsonrpc":"2.0","id":2,"method":"workspace/configuration","params":{"items":[{"section":"texlab"}]}}
2024-01-09T11:45:34.885 helix_term::application [DEBUG] received editor event: LanguageServerMessage((0, MethodCall(MethodCall { jsonrpc: Some(V2), method: "workspace/configuration", params: Map({"items": Array [Object {"section": String("texlab")}]}), id: Num(2) })))
2024-01-09T11:45:34.885 helix_lsp::transport [INFO] texlab -> {"jsonrpc":"2.0","result":[{"build":{"args":["-pdf","-interaction=nonstopmode","-synctex=1","%f"],"executable":"latexmk","forwardSearchAfter":true,"onSave":true},"chktex":{"onEdit":true,"onOpenAndSave":true},"forwardSearch":{"args":["--synctex-forward","%l:1:%f","%p"],"executable":"zathura"}}],"id":1}
2024-01-09T11:45:34.885 helix_lsp::transport [INFO] texlab -> {"jsonrpc":"2.0","result":[{"build":{"args":["-pdf","-interaction=nonstopmode","-synctex=1","%f"],"executable":"latexmk","forwardSearchAfter":true,"onSave":true},"chktex":{"onEdit":true,"onOpenAndSave":true},"forwardSearch":{"args":["--synctex-forward","%l:1:%f","%p"],"executable":"zathura"}}],"id":2}
2024-01-09T11:45:34.888 helix_lsp::transport [INFO] scls <- {"jsonrpc":"2.0","result":{"capabilities":{"completionProvider":{},"textDocumentSync":2}},"id":0}
2024-01-09T11:45:34.888 helix_lsp::transport [INFO] scls <- {"capabilities":{"completionProvider":{},"textDocumentSync":2}}
2024-01-09T11:45:34.888 helix_lsp::transport [INFO] scls -> {"jsonrpc":"2.0","method":"initialized","params":{}}
2024-01-09T11:45:34.888 helix_term::application [DEBUG] received editor event: LanguageServerMessage((2, Notification(Notification { jsonrpc: None, method: "initialized", params: None })))
2024-01-09T11:45:34.888 helix_lsp::transport [INFO] scls -> {"jsonrpc":"2.0","method":"workspace/didChangeConfiguration","params":{"settings":{"max-completion_items":20,"snippets_first":true}}}
2024-01-09T11:45:34.888 helix_lsp::transport [INFO] scls -> {"jsonrpc":"2.0","method":"textDocument/didOpen","params":{"textDocument":{"languageId":"latex","text":"\\documentclass[11pt]{article}\n\n\\begin{document}\n\tSample \\ref{fig:lorem} Figure\n\t\\begin{figure}\n\t\t\\caption{Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.}\\label{fig:lorem}\n\t\\end{figure}\n\\end{document}\n","uri":"file:///home/nohandrick/tmp/test.tex","version":0}}}
2024-01-09T11:45:34.888 helix_lsp::transport [INFO] scls <- {"jsonrpc":"2.0","method":"window/logMessage","params":{"message":"server initialized!","type":3}}
2024-01-09T11:45:34.888 helix_term::application [DEBUG] received editor event: LanguageServerMessage((2, Notification(Notification { jsonrpc: Some(V2), method: "window/logMessage", params: Map({"message": String("server initialized!"), "type": Number(3)}) })))
2024-01-09T11:45:34.888 helix_term::application [INFO] window/logMessage: LogMessageParams { typ: Info, message: "server initialized!" }
2024-01-09T11:45:34.888 helix_lsp::transport [INFO] scls <- {"jsonrpc":"2.0","method":"window/logMessage","params":{"message":"Did change configuration: DidChangeConfigurationParams { settings: Object {\"max-completion_items\": Number(20), \"snippets_first\": Bool(true)} }","type":3}}
2024-01-09T11:45:34.888 helix_lsp::transport [INFO] scls <- {"jsonrpc":"2.0","method":"window/logMessage","params":{"message":"Did open: file:///home/nohandrick/tmp/test.tex","type":3}}
2024-01-09T11:45:34.888 helix_term::application [DEBUG] received editor event: LanguageServerMessage((2, Notification(Notification { jsonrpc: Some(V2), method: "window/logMessage", params: Map({"message": String("Did change configuration: DidChangeConfigurationParams { settings: Object {\"max-completion_items\": Number(20), \"snippets_first\": Bool(true)} }"), "type": Number(3)}) })))
2024-01-09T11:45:34.888 helix_term::application [INFO] window/logMessage: LogMessageParams { typ: Info, message: "Did change configuration: DidChangeConfigurationParams { settings: Object {\"max-completion_items\": Number(20), \"snippets_first\": Bool(true)} }" }
2024-01-09T11:45:34.888 helix_term::application [DEBUG] received editor event: LanguageServerMessage((2, Notification(Notification { jsonrpc: Some(V2), method: "window/logMessage", params: Map({"message": String("Did open: file:///home/nohandrick/tmp/test.tex"), "type": Number(3)}) })))
2024-01-09T11:45:34.888 helix_term::application [INFO] window/logMessage: LogMessageParams { typ: Info, message: "Did open: file:///home/nohandrick/tmp/test.tex" }
2024-01-09T11:45:34.920 helix_term::application [DEBUG] received editor event: Redraw
2024-01-09T11:45:35.095 helix_lsp::transport [ERROR] ltex-ls err <- "SLF4J: Failed to load class \"org.slf4j.impl.StaticLoggerBinder\".\n"
2024-01-09T11:45:35.095 helix_lsp::transport [ERROR] ltex-ls err <- "SLF4J: Defaulting to no-operation (NOP) logger implementation\n"
2024-01-09T11:45:35.095 helix_lsp::transport [ERROR] ltex-ls err <- "SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.\n"
2024-01-09T11:45:35.187 helix_lsp::transport [INFO] texlab <- {"jsonrpc":"2.0","method":"textDocument/publishDiagnostics","params":{"diagnostics":[],"uri":"file:///home/nohandrick/tmp/test.tex"}}
2024-01-09T11:45:35.187 helix_lsp::transport [INFO] texlab <- {"jsonrpc":"2.0","method":"textDocument/publishDiagnostics","params":{"diagnostics":[],"uri":"file:///home/nohandrick/tmp/test.log"}}
2024-01-09T11:45:35.187 helix_lsp::transport [INFO] texlab <- {"jsonrpc":"2.0","method":"textDocument/publishDiagnostics","params":{"diagnostics":[],"uri":"file:///home/nohandrick/tmp/test.aux"}}
2024-01-09T11:45:35.187 helix_lsp::transport [INFO] texlab <- {"jsonrpc":"2.0","method":"textDocument/publishDiagnostics","params":{"diagnostics":[],"uri":"file:///home/nohandrick/tmp/text.tex"}}
2024-01-09T11:45:35.187 helix_term::application [DEBUG] received editor event: LanguageServerMessage((0, Notification(Notification { jsonrpc: Some(V2), method: "textDocument/publishDiagnostics", params: Map({"diagnostics": Array [], "uri": String("file:///home/nohandrick/tmp/test.tex")}) })))
2024-01-09T11:45:35.187 helix_term::application [DEBUG] received editor event: LanguageServerMessage((0, Notification(Notification { jsonrpc: Some(V2), method: "textDocument/publishDiagnostics", params: Map({"diagnostics": Array [], "uri": String("file:///home/nohandrick/tmp/test.log")}) })))
2024-01-09T11:45:35.187 helix_term::application [DEBUG] received editor event: LanguageServerMessage((0, Notification(Notification { jsonrpc: Some(V2), method: "textDocument/publishDiagnostics", params: Map({"diagnostics": Array [], "uri": String("file:///home/nohandrick/tmp/test.aux")}) })))
2024-01-09T11:45:35.187 helix_term::application [DEBUG] received editor event: LanguageServerMessage((0, Notification(Notification { jsonrpc: Some(V2), method: "textDocument/publishDiagnostics", params: Map({"diagnostics": Array [], "uri": String("file:///home/nohandrick/tmp/text.tex")}) })))
2024-01-09T11:45:35.221 helix_term::application [DEBUG] received editor event: Redraw
2024-01-09T11:45:37.694 helix_lsp::transport [ERROR] ltex-ls err <- "Jan 09, 2024 11:45:37 AM org.bsplines.ltexls.server.LtexLanguageServer initialize\n"
2024-01-09T11:45:37.694 helix_lsp::transport [ERROR] ltex-ls err <- "INFO: ltex-ls 16.0.0 - initializing...\n"
2024-01-09T11:45:37.704 helix_lsp::transport [INFO] ltex-ls <- {"jsonrpc":"2.0","id":0,"result":{"capabilities":{"textDocumentSync":1,"completionProvider":{},"codeActionProvider":{"codeActionKinds":["quickfix.ltex.acceptSuggestions"]},"executeCommandProvider":{"commands":["_ltex.checkDocument","_ltex.getServerStatus"]}}}}
2024-01-09T11:45:37.704 helix_lsp::transport [INFO] ltex-ls <- {"capabilities":{"codeActionProvider":{"codeActionKinds":["quickfix.ltex.acceptSuggestions"]},"completionProvider":{},"executeCommandProvider":{"commands":["_ltex.checkDocument","_ltex.getServerStatus"]},"textDocumentSync":1}}
2024-01-09T11:45:37.704 helix_lsp::transport [INFO] ltex-ls -> {"jsonrpc":"2.0","method":"initialized","params":{}}
2024-01-09T11:45:37.704 helix_term::application [DEBUG] received editor event: LanguageServerMessage((1, Notification(Notification { jsonrpc: None, method: "initialized", params: None })))
2024-01-09T11:45:37.704 helix_lsp::transport [INFO] ltex-ls -> {"jsonrpc":"2.0","method":"workspace/didChangeConfiguration","params":{"settings":{"ltex":{"dictionary":{censored}}}}}
2024-01-09T11:45:37.704 helix_lsp::transport [INFO] ltex-ls -> {"jsonrpc":"2.0","method":"textDocument/didOpen","params":{"textDocument":{"languageId":"latex","text":"\\documentclass[11pt]{article}\n\n\\begin{document}\n\tSample \\ref{fig:lorem} Figure\n\t\\begin{figure}\n\t\t\\caption{Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.}\\label{fig:lorem}\n\t\\end{figure}\n\\end{document}\n","uri":"file:///home/nohandrick/tmp/test.tex","version":0}}}
2024-01-09T11:45:37.708 helix_lsp::transport [INFO] ltex-ls <- {"jsonrpc":"2.0","id":"1","method":"window/workDoneProgress/create","params":{"token":"{\"uri\":\"file:///home/nohandrick/tmp/test.tex\",\"operation\":\"checkDocument\",\"uuid\":\"72dc0b20-c5b7-7c9f-41e8-d73409133090\"}"}}
2024-01-09T11:45:37.708 helix_term::application [DEBUG] received editor event: LanguageServerMessage((1, MethodCall(MethodCall { jsonrpc: Some(V2), method: "window/workDoneProgress/create", params: Map({"token": String("{\"uri\":\"file:///home/nohandrick/tmp/test.tex\",\"operation\":\"checkDocument\",\"uuid\":\"72dc0b20-c5b7-7c9f-41e8-d73409133090\"}")}), id: Str("1") })))
2024-01-09T11:45:37.708 helix_lsp::transport [INFO] ltex-ls -> {"jsonrpc":"2.0","result":null,"id":"1"}
2024-01-09T11:45:37.710 helix_lsp::transport [INFO] ltex-ls <- {"jsonrpc":"2.0","method":"$/progress","params":{"token":"{\"uri\":\"file:///home/nohandrick/tmp/test.tex\",\"operation\":\"checkDocument\",\"uuid\":\"72dc0b20-c5b7-7c9f-41e8-d73409133090\"}","value":{"kind":"begin","title":"Checking document","cancellable":false,"message":"file:///home/nohandrick/tmp/test.tex"}}}
2024-01-09T11:45:37.710 helix_term::application [DEBUG] received editor event: LanguageServerMessage((1, Notification(Notification { jsonrpc: Some(V2), method: "$/progress", params: Map({"token": String("{\"uri\":\"file:///home/nohandrick/tmp/test.tex\",\"operation\":\"checkDocument\",\"uuid\":\"72dc0b20-c5b7-7c9f-41e8-d73409133090\"}"), "value": Object {"cancellable": Bool(false), "kind": String("begin"), "message": String("file:///home/nohandrick/tmp/test.tex"), "title": String("Checking document")}}) })))
2024-01-09T11:45:37.710 helix_view::editor [DEBUG] editor status: [{"uri":"file:///home/nohandrick/tmp/test.tex","operation":"checkDocument","uuid":"72dc0b20-c5b7-7c9f-41e8-d73409133090"}] Checking document - file:///home/nohandrick/tmp/test.tex
2024-01-09T11:45:37.710 helix_lsp::transport [INFO] ltex-ls <- {"jsonrpc":"2.0","id":"2","method":"workspace/configuration","params":{"items":[{"scopeUri":"file:///home/nohandrick/tmp/test.tex","section":"ltex"}]}}
2024-01-09T11:45:37.710 helix_term::application [DEBUG] received editor event: LanguageServerMessage((1, MethodCall(MethodCall { jsonrpc: Some(V2), method: "workspace/configuration", params: Map({"items": Array [Object {"scopeUri": String("file:///home/nohandrick/tmp/test.tex"), "section": String("ltex")}]}), id: Str("2") })))
2024-01-09T11:45:37.710 helix_lsp::transport [INFO] ltex-ls -> {"jsonrpc":"2.0","result":[{"dictionary":{censored}}],"id":"2"}
2024-01-09T11:45:37.738 helix_term::application [DEBUG] received editor event: Redraw
2024-01-09T11:45:37.750 helix_lsp::transport [ERROR] ltex-ls err <- "Jan 09, 2024 11:45:37 AM org.bsplines.ltexls.server.DocumentChecker logTextToBeChecked\n"
2024-01-09T11:45:37.750 helix_lsp::transport [ERROR] ltex-ls err <- "FINE: Checking the following text in language 'en-US' via LanguageTool: \"\\n\\nSample Dummy0 Figure Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lo\"... (truncated to 100 characters)\n"
2024-01-09T11:45:38.760 helix_lsp::transport [ERROR] ltex-ls err <- "Jan 09, 2024 11:45:38 AM org.bsplines.ltexls.server.DocumentChecker checkAnnotatedTextFragment\n"
2024-01-09T11:45:38.760 helix_lsp::transport [ERROR] ltex-ls err <- "FINE: Obtained 12 rule matches\n"
2024-01-09T11:45:38.760 helix_lsp::transport [INFO] ltex-ls <- {"jsonrpc":"2.0","method":"$/progress","params":{"token":"{\"uri\":\"file:///home/nohandrick/tmp/test.tex\",\"operation\":\"checkDocument\",\"uuid\":\"72dc0b20-c5b7-7c9f-41e8-d73409133090\"}","value":{"kind":"end"}}}
2024-01-09T11:45:38.760 helix_term::application [DEBUG] received editor event: LanguageServerMessage((1, Notification(Notification { jsonrpc: Some(V2), method: "$/progress", params: Map({"token": String("{\"uri\":\"file:///home/nohandrick/tmp/test.tex\",\"operation\":\"checkDocument\",\"uuid\":\"72dc0b20-c5b7-7c9f-41e8-d73409133090\"}"), "value": Object {"kind": String("end")}}) })))
2024-01-09T11:45:38.765 helix_lsp::transport [INFO] ltex-ls <- {"jsonrpc":"2.0","method":"textDocument/publishDiagnostics","params":{"uri":"file:///home/nohandrick/tmp/test.tex","diagnostics":[{"range":{"start":{"line":5,"character":11},"end":{"line":5,"character":16}},"severity":3,"code":"MORFOLOGIK_RULE_EN_US","codeDescription":{"href":"https://community.languagetool.org/rule/show/MORFOLOGIK_RULE_EN_US?lang\u003den-US"},"source":"LTeX","message":"\u0027Lorem\u0027: Possible spelling mistake found."},{"range":{"start":{"line":5,"character":17},"end":{"line":5,"character":22}},"severity":3,"code":"MORFOLOGIK_RULE_EN_US","codeDescription":{"href":"https://community.languagetool.org/rule/show/MORFOLOGIK_RULE_EN_US?lang\u003den-US"},"source":"LTeX","message":"\u0027Ipsum\u0027: Possible spelling mistake found."},{"range":{"start":{"line":5,"character":86},"end":{"line":5,"character":91}},"severity":3,"code":"MORFOLOGIK_RULE_EN_US","codeDescription":{"href":"https://community.languagetool.org/rule/show/MORFOLOGIK_RULE_EN_US?lang\u003den-US"},"source":"LTeX","message":"\u0027Lorem\u0027: Possible spelling mistake found."},{"range":{"start":{"line":5,"character":92},"end":{"line":5,"character":97}},"severity":3,"code":"MORFOLOGIK_RULE_EN_US","codeDescription":{"href":"https://community.languagetool.org/rule/show/MORFOLOGIK_RULE_EN_US?lang\u003den-US"},"source":"LTeX","message":"\u0027Ipsum\u0027: Possible spelling mistake found."},{"range":{"start":{"line":5,"character":385},"end":{"line":5,"character":396}},"severity":3,"code":"MORFOLOGIK_RULE_EN_US","codeDescription":{"href":"https://community.languagetool.org/rule/show/MORFOLOGIK_RULE_EN_US?lang\u003den-US"},"source":"LTeX","message":"\u0027popularised\u0027: Possible spelling mistake. \u0027popularised\u0027 is British English."},{"range":{"start":{"line":5,"character":430},"end":{"line":5,"character":438}},"severity":3,"code":"MORFOLOGIK_RULE_EN_US","codeDescription":{"href":"https://community.languagetool.org/rule/show/MORFOLOGIK_RULE_EN_US?lang\u003den-US"},"source":"LTeX","message":"\u0027Letraset\u0027: Possible spelling mistake found."},{"range":{"start":{"line":5,"character":457},"end":{"line":5,"character":462}},"severity":3,"code":"MORFOLOGIK_RULE_EN_US","codeDescription":{"href":"https://community.languagetool.org/rule/show/MORFOLOGIK_RULE_EN_US?lang\u003den-US"},"source":"LTeX","message":"\u0027Lorem\u0027: Possible spelling mistake found."},{"range":{"start":{"line":5,"character":463},"end":{"line":5,"character":468}},"severity":3,"code":"MORFOLOGIK_RULE_EN_US","codeDescription":{"href":"https://community.languagetool.org/rule/show/MORFOLOGIK_RULE_EN_US?lang\u003den-US"},"source":"LTeX","message":"\u0027Ipsum\u0027: Possible spelling mistake found."},{"range":{"start":{"line":5,"character":535},"end":{"line":5,"character":540}},"severity":3,"code":"MORFOLOGIK_RULE_EN_US","codeDescription":{"href":"https://community.languagetool.org/rule/show/MORFOLOGIK_RULE_EN_US?lang\u003den-US"},"source":"LTeX","message":"\u0027Aldus\u0027: Possible spelling mistake found."},{"range":{"start":{"line":5,"character":541},"end":{"line":5,"character":550}},"severity":3,"code":"MORFOLOGIK_RULE_EN_US","codeDescription":{"href":"https://community.languagetool.org/rule/show/MORFOLOGIK_RULE_EN_US?lang\u003den-US"},"source":"LTeX","message":"\u0027PageMaker\u0027: Possible spelling mistake found."},{"range":{"start":{"line":5,"character":573},"end":{"line":5,"character":578}},"severity":3,"code":"MORFOLOGIK_RULE_EN_US","codeDescription":{"href":"https://community.languagetool.org/rule/show/MORFOLOGIK_RULE_EN_US?lang\u003den-US"},"source":"LTeX","message":"\u0027Lorem\u0027: Possible spelling mistake found."},{"range":{"start":{"line":5,"character":579},"end":{"line":5,"character":584}},"severity":3,"code":"MORFOLOGIK_RULE_EN_US","codeDescription":{"href":"https://community.languagetool.org/rule/show/MORFOLOGIK_RULE_EN_US?lang\u003den-US"},"source":"LTeX","message":"\u0027Ipsum\u0027: Possible spelling mistake found."}]}}
2024-01-09T11:45:38.765 helix_term::application [DEBUG] received editor event: LanguageServerMessage((1, Notification(Notification { jsonrpc: Some(V2), method: "textDocument/publishDiagnostics", params: Map({"diagnostics": Array [Object {"code": String("MORFOLOGIK_RULE_EN_US"), "codeDescription": Object {"href": String("https://community.languagetool.org/rule/show/MORFOLOGIK_RULE_EN_US?lang=en-US")}, "message": String("'Lorem': Possible spelling mistake found."), "range": Object {"end": Object {"character": Number(16), "line": Number(5)}, "start": Object {"character": Number(11), "line": Number(5)}}, "severity": Number(3), "source": String("LTeX")}, Object {"code": String("MORFOLOGIK_RULE_EN_US"), "codeDescription": Object {"href": String("https://community.languagetool.org/rule/show/MORFOLOGIK_RULE_EN_US?lang=en-US")}, "message": String("'Ipsum': Possible spelling mistake found."), "range": Object {"end": Object {"character": Number(22), "line": Number(5)}, "start": Object {"character": Number(17), "line": Number(5)}}, "severity": Number(3), "source": String("LTeX")}, Object {"code": String("MORFOLOGIK_RULE_EN_US"), "codeDescription": Object {"href": String("https://community.languagetool.org/rule/show/MORFOLOGIK_RULE_EN_US?lang=en-US")}, "message": String("'Lorem': Possible spelling mistake found."), "range": Object {"end": Object {"character": Number(91), "line": Number(5)}, "start": Object {"character": Number(86), "line": Number(5)}}, "severity": Number(3), "source": String("LTeX")}, Object {"code": String("MORFOLOGIK_RULE_EN_US"), "codeDescription": Object {"href": String("https://community.languagetool.org/rule/show/MORFOLOGIK_RULE_EN_US?lang=en-US")}, "message": String("'Ipsum': Possible spelling mistake found."), "range": Object {"end": Object {"character": Number(97), "line": Number(5)}, "start": Object {"character": Number(92), "line": Number(5)}}, "severity": Number(3), "source": String("LTeX")}, Object {"code": String("MORFOLOGIK_RULE_EN_US"), "codeDescription": Object {"href": String("https://community.languagetool.org/rule/show/MORFOLOGIK_RULE_EN_US?lang=en-US")}, "message": String("'popularised': Possible spelling mistake. 'popularised' is British English."), "range": Object {"end": Object {"character": Number(396), "line": Number(5)}, "start": Object {"character": Number(385), "line": Number(5)}}, "severity": Number(3), "source": String("LTeX")}, Object {"code": String("MORFOLOGIK_RULE_EN_US"), "codeDescription": Object {"href": String("https://community.languagetool.org/rule/show/MORFOLOGIK_RULE_EN_US?lang=en-US")}, "message": String("'Letraset': Possible spelling mistake found."), "range": Object {"end": Object {"character": Number(438), "line": Number(5)}, "start": Object {"character": Number(430), "line": Number(5)}}, "severity": Number(3), "source": String("LTeX")}, Object {"code": String("MORFOLOGIK_RULE_EN_US"), "codeDescription": Object {"href": String("https://community.languagetool.org/rule/show/MORFOLOGIK_RULE_EN_US?lang=en-US")}, "message": String("'Lorem': Possible spelling mistake found."), "range": Object {"end": Object {"character": Number(462), "line": Number(5)}, "start": Object {"character": Number(457), "line": Number(5)}}, "severity": Number(3), "source": String("LTeX")}, Object {"code": String("MORFOLOGIK_RULE_EN_US"), "codeDescription": Object {"href": String("https://community.languagetool.org/rule/show/MORFOLOGIK_RULE_EN_US?lang=en-US")}, "message": String("'Ipsum': Possible spelling mistake found."), "range": Object {"end": Object {"character": Number(468), "line": Number(5)}, "start": Object {"character": Number(463), "line": Number(5)}}, "severity": Number(3), "source": String("LTeX")}, Object {"code": String("MORFOLOGIK_RULE_EN_US"), "codeDescription": Object {"href": String("https://community.languagetool.org/rule/show/MORFOLOGIK_RULE_EN_US?lang=en-US")}, "message": String("'Aldus': Possible spelling mistake found."), "range": Object {"end": Object {"character": Number(540), "line": Number(5)}, "start": Object {"character": Number(535), "line": Number(5)}}, "severity": Number(3), "source": String("LTeX")}, Object {"code": String("MORFOLOGIK_RULE_EN_US"), "codeDescription": Object {"href": String("https://community.languagetool.org/rule/show/MORFOLOGIK_RULE_EN_US?lang=en-US")}, "message": String("'PageMaker': Possible spelling mistake found."), "range": Object {"end": Object {"character": Number(550), "line": Number(5)}, "start": Object {"character": Number(541), "line": Number(5)}}, "severity": Number(3), "source": String("LTeX")}, Object {"code": String("MORFOLOGIK_RULE_EN_US"), "codeDescription": Object {"href": String("https://community.languagetool.org/rule/show/MORFOLOGIK_RULE_EN_US?lang=en-US")}, "message": String("'Lorem': Possible spelling mistake found."), "range": Object {"end": Object {"character": Number(578), "line": Number(5)}, "start": Object {"character": Number(573), "line": Number(5)}}, "severity": Number(3), "source": String("LTeX")}, Object {"code": String("MORFOLOGIK_RULE_EN_US"), "codeDescription": Object {"href": String("https://community.languagetool.org/rule/show/MORFOLOGIK_RULE_EN_US?lang=en-US")}, "message": String("'Ipsum': Possible spelling mistake found."), "range": Object {"end": Object {"character": Number(584), "line": Number(5)}, "start": Object {"character": Number(579), "line": Number(5)}}, "severity": Number(3), "source": String("LTeX")}], "uri": String("file:///home/nohandrick/tmp/test.tex")}) })))
2024-01-09T11:45:38.794 helix_term::application [DEBUG] received editor event: Redraw
2024-01-09T11:45:43.106 helix_term::application [DEBUG] received editor event: IdleTimer
2024-01-09T11:45:43.106 helix_lsp::transport [INFO] texlab -> {"jsonrpc":"2.0","method":"textDocument/inlayHint","params":{"range":{"end":{"character":0,"line":8},"start":{"character":0,"line":0}},"textDocument":{"uri":"file:///home/nohandrick/tmp/test.tex"}},"id":1}
2024-01-09T11:45:43.106 helix_lsp::transport [INFO] texlab <- {"jsonrpc":"2.0","id":1,"result":[{"label":" Figure 1: Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum. ","paddingLeft":true,"position":{"character":23,"line":3}},{"label":" Figure 1 ","paddingLeft":true,"position":{"character":603,"line":5}}]}
2024-01-09T11:45:43.106 helix_lsp::transport [INFO] texlab <- [{"label":" Figure 1: Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum. ","paddingLeft":true,"position":{"character":23,"line":3}},{"label":" Figure 1 ","paddingLeft":true,"position":{"character":603,"line":5}}]
2024-01-09T11:45:43.531 helix_term::application [DEBUG] received editor event: IdleTimer
2024-01-09T11:45:43.711 helix_term::application [DEBUG] received editor event: IdleTimer
2024-01-09T11:45:43.872 helix_term::application [DEBUG] received editor event: IdleTimer
2024-01-09T11:45:44.031 helix_term::application [DEBUG] received editor event: IdleTimer
2024-01-09T11:45:44.204 helix_term::application [DEBUG] received editor event: IdleTimer
2024-01-09T11:45:44.477 helix_term::application [DEBUG] received editor event: IdleTimer
2024-01-09T11:45:44.662 helix_term::application [DEBUG] received editor event: IdleTimer
2024-01-09T11:45:44.816 helix_term::application [DEBUG] received editor event: IdleTimer
2024-01-09T11:45:44.974 helix_term::application [DEBUG] received editor event: IdleTimer
2024-01-09T11:45:45.122 helix_term::application [DEBUG] received editor event: IdleTimer
2024-01-09T11:45:45.284 helix_term::application [DEBUG] received editor event: IdleTimer
2024-01-09T11:45:45.432 helix_term::application [DEBUG] received editor event: IdleTimer
2024-01-09T11:45:46.137 helix_term::application [DEBUG] received editor event: IdleTimer
2024-01-09T11:45:46.289 helix_term::commands::typed [DEBUG] quitting...
2024-01-09T11:45:46.469 helix_term::commands::typed [DEBUG] quitting...
2024-01-09T11:45:46.469 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 0
2024-01-09T11:45:46.469 helix_term::job [DEBUG] waiting on jobs...
2024-01-09T11:45:46.469 helix_term::job [DEBUG] waiting on jobs...
2024-01-09T11:45:46.469 helix_lsp::file_event [DEBUG] Removing LSP client: 0
2024-01-09T11:45:46.469 helix_lsp::transport [INFO] texlab -> {"jsonrpc":"2.0","method":"shutdown","id":2}
2024-01-09T11:45:46.469 helix_lsp::file_event [DEBUG] Removing LSP client: 1
2024-01-09T11:45:46.469 helix_lsp::file_event [DEBUG] Removing LSP client: 2
2024-01-09T11:45:46.469 helix_lsp::transport [INFO] scls -> {"jsonrpc":"2.0","method":"shutdown","id":1}
2024-01-09T11:45:46.469 helix_lsp::transport [INFO] ltex-ls -> {"jsonrpc":"2.0","method":"shutdown","id":1}
2024-01-09T11:45:46.469 helix_lsp::transport [INFO] texlab <- {"jsonrpc":"2.0","id":2,"result":null}
2024-01-09T11:45:46.469 helix_lsp::transport [INFO] texlab <- null
2024-01-09T11:45:46.469 helix_lsp::transport [INFO] texlab -> {"jsonrpc":"2.0","method":"exit"}
2024-01-09T11:45:46.469 helix_lsp::transport [INFO] scls <- {"jsonrpc":"2.0","result":null,"id":1}
2024-01-09T11:45:46.469 helix_lsp::transport [INFO] scls <- null
2024-01-09T11:45:46.469 helix_lsp::transport [INFO] scls -> {"jsonrpc":"2.0","method":"exit"}
2024-01-09T11:45:46.470 helix_lsp::transport [ERROR] ltex-ls err <- "Jan 09, 2024 11:45:46 AM org.bsplines.ltexls.server.LtexLanguageServer shutdown\n"
2024-01-09T11:45:46.470 helix_lsp::transport [ERROR] ltex-ls err <- "INFO: Shutting down ltex-ls...\n"
2024-01-09T11:45:46.470 helix_lsp::transport [INFO] ltex-ls <- {"jsonrpc":"2.0","id":1,"result":{}}
2024-01-09T11:45:46.470 helix_lsp::transport [INFO] ltex-ls <- {}
2024-01-09T11:45:46.470 helix_lsp::client [WARN] language server failed to terminate gracefully - failed to parse: invalid type: map, expected unit
2024-01-09T11:45:46.470 helix_lsp::transport [INFO] ltex-ls -> {"jsonrpc":"2.0","method":"exit"}

```

</details>

### Platform

Linux

### Terminal Emulator

alacritty

### Installation Method

Arch AUR: helix-git (btw)

### Helix Version

helix 23.10 (783ff27b)

**Comment by pascalkuthe** [comment]
I noticted this too in latex. Qutie annoying, I will try to fix this in #6417 when I get that rebased again.

duplicate of #9218
